modules/escalaxBidAdapter.js

```diff
@@ -0,0 +1,84 @@
+import { ortbConverter } from '../libraries/ortbConverter/converter.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { config } from '../src/config.js';
+import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js';
+
+const BIDDER_CODE = 'escalax';
+const ESCALAX_SOURCE_ID_MACRO = '[sourceId]';
+const ESCALAX_ACCOUNT_ID_MACRO = '[accountId]';
+const ESCALAX_SUBDOMAIN_MACRO = '[subdomain]';
+const ESCALAX_URL = `https://${ESCALAX_SUBDOMAIN_MACRO}.escalax.io/bid?type=pjs&partner=${ESCALAX_SOURCE_ID_MACRO}&token=${ESCALAX_ACCOUNT_ID_MACRO}`;
+const ESCALAX_DEFAULT_CURRENCY = 'USD';
+const ESCALAX_DEFAULT_SUBDOMAIN = 'bidder_us';
+
+function createImp(buildImp, bidRequest, context) {
+  const imp = buildImp(bidRequest, context);
+  imp.ext = {
+    [BIDDER_CODE]: {
+      sourceId: bidRequest.params.sourceId,
+      accountId: bidRequest.params.accountId,
+      subdomain: bidRequest.params.subdomain || ESCALAX_DEFAULT_SUBDOMAIN,
+    }
+  };
+  if (!imp.bidfloor) imp.bidfloor = bidRequest.params.bidfloor;
+  return imp;
+}
+
+function createRequest(buildRequest, imps, bidderRequest, context) {
+  const request = buildRequest(imps, bidderRequest, context);
+  const bid = context.bidRequests[0];
+  request.test = config.getConfig('debug') ? 1 : 0;
+  if (!request.cur) request.cur = [bid.params.currency || ESCALAX_DEFAULT_CURRENCY];
+  return request;
+}
+
+function createBidResponse(buildBidResponse, bid, context) {
+  const bidResponse = buildBidResponse(bid, context);
+  bidResponse.cur = 'USD';
+  return bidResponse;
+}
+
+const converter = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: 20,
+  },
+  imp: createImp,
+  request: createRequest,
+  bidResponse: createBidResponse
+});
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER, VIDEO, NATIVE],
+
+  isBidRequestValid: (bid) => {
+    return Boolean(bid.params.sourceId) && Boolean(bid.params.accountId);
+  },
+
+  buildRequests: (validBidRequests, bidderRequest) => {
+    if (validBidRequests && validBidRequests.length === 0) return [];
+    const { sourceId, accountId } = validBidRequests[0].params;
+    const subdomain = validBidRequests[0].params.subdomain;
```

Comment:
here, you use the subdomain of the first bid for the request.
So if bids uses different subdomain, it would be ignored.
I am guessing you want the publisher to use only one value for subdomain among all bids but this forces the publisher to copy paste the same parameter everywhere.
Another solution could be to set up a parameter in the setConfig/setBidderConfig option so the publisher would only have to set the param once and there would not an issue where only the params from the first bid is used. 
You can find a simple example of that with Adot bidder which setup a global publisherId : 
https://github.com/prebid/Prebid.js/blob/master/modules/adotBidAdapter.md
https://github.com/prebid/Prebid.js/blob/master/modules/adotBidAdapter.js

Reply:
@JulieLorin I see, that's reasonable
But wouldn't it be more difficult for the publisher to specify an additional code like in adot example with this `publisherId`
And what i saw in couple of adapters is `region` parameter, like here:
https://github.com/prebid/Prebid.js/blob/39e6418bb05501dbe1180bda44865c6b0ba938f2/modules/luceadBidAdapter.md?plain=1#L26-L40
or here:
https://github.com/prebid/Prebid.js/blob/39e6418bb05501dbe1180bda44865c6b0ba938f2/modules/precisoBidAdapter.md?plain=1#L15-L19

We can agree to make a `region` parameter instead of `subdomain` and a different endpoint url for each such parameter - US, EU

Reply:
why don't you detect the user region and fill in your own subdomain instead of putting this on the publisher. Here is some basic logic:

```
function getContinentFromTimeZone(timeZone) {
    // Validate the input timezone
    if (typeof timeZone !== 'string') {
        throw new Error('Invalid timezone: Timezone must be a string.');
    }

    // Map timezone region prefixes to continents
    const continentMap = {
        "Africa": "Africa",
        "America": "North America / South America",
        "Antarctica": "Antarctica",
        "Asia": "Asia",
        "Atlantic": "Atlantic Ocean region",
        "Australia": "Australia",
        "Europe": "Europe",
        "Indian": "Indian Ocean region",
        "Pacific": "Pacific Ocean region",
    };

    // Extract the main region from the timezone string
    const mainRegion = timeZone.split('/')[0];

    // Return the matching continent or a default message
    return continentMap[mainRegion] || "Unknown continent";
}

// Example usage:
const userTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
console.log(`User's timezone: ${userTimeZone}`);
console.log(`Continent: ${getContinentFromTimeZone(userTimeZone)}`);
```

Reply:
Looks like a good solution to me, thanks!

Reply:
@JulieLorin what do you think?